modules/nf-core/gmmdemux/main.nf

```diff
@@ -0,0 +1,71 @@
+
+process GMMDEMUX {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gmm-demux:0.2.2.3--pyh7cba7a3_0':
+        'biocontainers/gmm-demux:0.2.2.3--pyh7cba7a3_0' }"
+
+    input:
+    tuple val(meta), path(cell_hashing_barcodes,stageAs: "hto_files/*"), path(cell_hashing_matrix,stageAs: "hto_files/*"),path(cell_hashing_features,stageAs: "hto_files/*"),val(hto_names)
+    val csv
+    val num_cells
+    val output_dir
+    val full_report
+    val simplified_report
+    val examine
+    val ambigous
+    val extract
+    val skip
+
+    output:
+    tuple val(meta), path('test/barcodes.tsv.gz'), emit: barcodes
+    tuple val(meta), path('test/*.mtx.gz'), emit: matrix
+    tuple val(meta), path('test/features.tsv.gz'), emit: features
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    //since this tool has many optional inputs that can be passed in, we need to check if they are null or not
+    // in order to produce or not certain reports or outputs
```

Comment:
These should all come via `ext.arg`, if they generate optional outputs, just set `optional: true` accordingly